checker.cpp

```diff
@@ -2,21 +2,42 @@
 #include <iostream>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+bool findIfNotInRange(float value, float min, float max)
+{
+    return (value < min || value > max);
+}
+
+bool checkTemperature(float temperature)
+{
+  bool result {true};
+  if(findIfNotInRange(temperature, 0, 45)) {
```

Comment:
usage of hard coded values will lead to issues during maintenance. if it gets used in many places and then a change in one place needs to be consistent with all others. There is a scope for improvement here.

Reply:
addressed